src/Domain.LinnApps/PurchaseOrderReqs/PurchaseOrderReqService.cs

```diff
@@ -80,11 +92,13 @@ public void Authorise(PurchaseOrderReq entity, IEnumerable<string> privileges, i
             {
                 entity.State = allowedToAuthoriseResult.NewState;
                 entity.AuthorisedById = currentUserId;
+                entity.AuthorisedBy = this.employeeRepository.FindById(currentUserId);
```

Comment:
from a quick look a lot of these DomainService if (..) statements don't have corresponding unit test for WhenThatScenario.. green ncrunch dots are your friend